src/Tokenizer/Tokens.php

```diff
@@ -314,7 +314,31 @@ public function offsetUnset($index): void
      */
     public function offsetSet($index, $newval): void
     {
+        // # cache usage count with following variant: tests/Fixer/: 102_074
+        // $this->blockStartCache = [];
+        // $this->blockEndCache = [];
+
         if (!isset($this[$index]) || !$this[$index]->equals($newval)) {
+            // # cache usage count with following variant: tests/Fixer/: 108_464
+            // $this->blockStartCache = [];
+            // $this->blockEndCache = [];
+
+            if (isset($this[$index]) && self::isTokenBlockEdge($this[$index])) {
```

Comment:
approach wasn't optimal, but there is still value behind it - extracted to https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7655